decoder.go

```diff
@@ -146,60 +179,71 @@ func (d *decoder) decode(nalu []byte) (image.Image, error) {
 	avPacket.size = C.int(len(nalu))
 	res := C.avcodec_send_packet(d.codecCtx, &avPacket)
 	if res < 0 {
+		//nolint:nilnil // avcodec_send_packet raises non-critical errors
```

Comment:
Something weird is happening with linter where it errors out on this nil nil now when it didn't before. Did we bump our lint deps or something?

Reply:
it always used to in my prs. I think it just ignored files with Cgo but may now include them. you can return an empty result struct if nil.

Reply:
I added a lint ignore with a TODO attached to a ticket to fix it later